File: CS/CustomLogonParametersExample.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
using CustomLogonParametersExample.Module.BusinessObjects;

namespace CustomLogonParametersExample.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            SecurityRole anonymousRole = ObjectSpace.FindObject<SecurityRole>(
                new BinaryOperator("Name", SecurityStrategy.AnonymousUserName));
            if (anonymousRole == null) {
                anonymousRole = ObjectSpace.CreateObject<SecurityRole>();
                anonymousRole.Name = SecurityStrategy.AnonymousUserName;
                anonymousRole.BeginUpdate();
                anonymousRole.Permissions[typeof(Company)].Grant(SecurityOperations.Read);
                anonymousRole.Permissions[typeof(Employee)].Grant(SecurityOperations.Read);
                anonymousRole.EndUpdate();
                anonymousRole.Save();
            }
            Employee anonymousUser = ObjectSpace.FindObject<Employee>(
                new BinaryOperator("UserName", SecurityStrategy.AnonymousUserName));
            if (anonymousUser == null) {
                anonymousUser = ObjectSpace.CreateObject<Employee>();
                anonymousUser.UserName = SecurityStrategy.AnonymousUserName;
                anonymousUser.IsActive = true;
                anonymousUser.SetPassword("");
                anonymousUser.Roles.Add(anonymousRole);
                anonymousUser.Save();
            }

            SecurityRole administratorRole = ObjectSpace.FindObject<SecurityRole>(
                new BinaryOperator("Name", SecurityStrategy.AdministratorRoleName));
            if (administratorRole == null) {
                administratorRole = ObjectSpace.CreateObject<SecurityRole>();
                administratorRole.Name = SecurityStrategy.AdministratorRoleName;
                administratorRole.CanEditModel = true;
            }
            administratorRole.BeginUpdate();
            administratorRole.Permissions.GrantRecursive(typeof(object), SecurityOperations.FullAccess);
            administratorRole.EndUpdate();
            administratorRole.Save();
            string adminName = "Administrator";
            Employee administratorUser = ObjectSpace.FindObject<Employee>(
                new BinaryOperator("UserName", adminName));
            if (administratorUser == null) {
                administratorUser = ObjectSpace.CreateObject<Employee>();
                administratorUser.UserName = adminName;
                administratorUser.IsActive = true;
                administratorUser.SetPassword("");
                administratorUser.Roles.Add(administratorRole);
                administratorUser.Save();
            }


            if (ObjectSpace.FindObject<Company>(null) == null) {
               
                Company company1 = ObjectSpace.CreateObject<Company>();
                company1.Name = "Company 1";
                company1.Employees.Add(administratorUser);
                company1.Save();

                Employee user1 = ObjectSpace.CreateObject<Employee>();
                user1.UserName = "Sam";
                user1.SetPassword("");
                user1.Save();

                Employee user2 = ObjectSpace.CreateObject<Employee>();
                user2.UserName = "John";
                user2.SetPassword("");
                user2.Save();

                Company company2 = ObjectSpace.CreateObject<Company>();
                company2.Name = "Company 2";
                company2.Employees.Add(user1);
                company2.Employees.Add(user2);
                company2.Save();

            }
            ObjectSpace.CommitChanges();
        }
    }
}

File: VB/CustomLogonParametersExample.Module/DatabaseUpdate/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security
Imports CustomLogonParametersExample.Module.BusinessObjects
Imports DevExpress.ExpressApp.Security.Strategy

Namespace CustomLogonParametersExample.Module.DatabaseUpdate
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim administrativeRole As SecuritySystemRole = ObjectSpace.FindObject(Of SecuritySystemRole)(New BinaryOperator("Name", SecurityStrategy.AdministratorRoleName))
			If administrativeRole Is Nothing Then
				administrativeRole = ObjectSpace.CreateObject(Of SecuritySystemRole)()
				administrativeRole.Name = SecurityStrategy.AdministratorRoleName
				administrativeRole.IsAdministrative = True
			End If
			Dim adminName As String = "Administrator"
			Dim administratorUser As Employee = ObjectSpace.FindObject(Of Employee)(New BinaryOperator("UserName", adminName))
			If administratorUser Is Nothing Then
				administratorUser = ObjectSpace.CreateObject(Of Employee)()
				administratorUser.UserName = adminName
				administratorUser.IsActive = True
				administratorUser.SetPassword("")
				administratorUser.Roles.Add(administrativeRole)
			End If
			Dim userRole As SecuritySystemRole = ObjectSpace.FindObject(Of SecuritySystemRole)(New BinaryOperator("Name", "User"))
			If userRole Is Nothing Then
				userRole = ObjectSpace.CreateObject(Of SecuritySystemRole)()
				userRole.Name = "User"
				Dim employeeReadOnlyPermission As SecuritySystemTypePermissionObject = ObjectSpace.CreateObject(Of SecuritySystemTypePermissionObject)()
				employeeReadOnlyPermission.TargetType = GetType(Employee)
				employeeReadOnlyPermission.AllowNavigate = True
				employeeReadOnlyPermission.AllowRead = True
				Dim conpanyReadOnlyPermission As SecuritySystemTypePermissionObject = ObjectSpace.CreateObject(Of SecuritySystemTypePermissionObject)()
				conpanyReadOnlyPermission.TargetType = GetType(Company)
				conpanyReadOnlyPermission.AllowNavigate = True
				conpanyReadOnlyPermission.AllowRead = True
				userRole.TypePermissions.Add(employeeReadOnlyPermission)
				userRole.TypePermissions.Add(conpanyReadOnlyPermission)
			End If
			If ObjectSpace.FindObject(Of Company)(Nothing) Is Nothing Then
				Dim company1 As Company = ObjectSpace.CreateObject(Of Company)()
				company1.Name = "Company 1"
				company1.Employees.Add(administratorUser)
				Dim user1 As Employee = ObjectSpace.CreateObject(Of Employee)()
				user1.UserName = "Sam"
				user1.SetPassword("")
				user1.Roles.Add(userRole)
				Dim user2 As Employee = ObjectSpace.CreateObject(Of Employee)()
				user2.UserName = "John"
				user2.SetPassword("")
				user2.Roles.Add(userRole)
				Dim company2 As Company = ObjectSpace.CreateObject(Of Company)()
				company2.Name = "Company 2"
				company2.Employees.Add(user1)
				company2.Employees.Add(user2)
			End If
		End Sub
	End Class
End Namespace
